test/coverage/datamodel/UuidSerializerTest.cpp

```diff
@@ -23,16 +27,235 @@ class TestFixture : public testing::Test {
 
 	// Run once per execution of the test application.
 	// Used for setup of all tests. Has access to this instance.
-	TestFixture() = default;
-	~TestFixture() = default;
+	TestUuidSerializer() = default;
+	~TestUuidSerializer() = default;
 
 	// Run once per execution of the test application.
 	// Used only for global setup outside of tests.
 	static void SetUpTestSuite() {}
 	static void TearDownTestSuite() {}
 };
 
-// TODO replace
-TEST_F(TestFixture, SomeTestName) {}
+// Test string serialization
+TEST_F(TestUuidSerializer, SerializeToString) {
+	uprotocol::v1::UUID uuid;
+	uuid.set_msb(0x1234567890ABCDEF);
+	uuid.set_lsb(0xFEDCBA0987654321);
+
+	std::string serialized_uuid =
+	    uprotocol::datamodel::serializer::uuid::AsString::serialize(uuid);
+	// Assert the serialized UUID matches the expected value
+	EXPECT_EQ(serialized_uuid, "12345678-90ab-cdef-fedc-ba0987654321");
```

Comment:
Makes me want to ask --- The code doesn't care about case .... does the spec specify lower vs upper case for hex digits?

Reply:
there are some other cases of mixed cases which should cover this

Reply:
The spec is case insensitive: https://www.rfc-editor.org/rfc/rfc9562.html#name-uuid-format